testsuite/mlir/mlir-parse/for/mlir.txt

```diff
@@ -0,0 +1,27 @@
+
+
+module {
+  func @apply(!type.unk) -> !type.S32
+  func @has_value(!type.unk) -> i1
+  func @next(!type.unk)
+  func @foo(!type.S32)
+  func @f(%arg0: !type.unk) {
+    %0 = "verona.alloca"() : () -> !type.alloca
+    %1 = "verona.store"(%arg0, %0) : (!type.unk, !type.alloca) -> !type.unk
+    %2 = "verona.alloca"() : () -> !type.alloca
+    %3 = "verona.constant(0)"() : () -> !type.int
+    %4 = "verona.store"(%3, %2) : (!type.int, !type.alloca) -> !type.unk
+    %5 = "verona.load"(%0) : (!type.alloca) -> !type.unk
+    br ^bb1
+  ^bb1:  // 2 preds: ^bb0, ^bb2
+    %6 = call @has_value(%5) : (!type.unk) -> i1
+    cond_br %6, ^bb2, ^bb3
+  ^bb2:  // pred: ^bb1
+    %7 = call @apply(%5) : (!type.unk) -> !type.S32
+    call @next(%5) : (!type.unk) -> ()
```

Comment:
Should `next` occur before or after the loop body? If it has side-effects then the two semantics are observable. Personally, I think after the body would lead to fewer confusions if it has side-effects.

Reply:
I see what you mean, and I agree. One example is logging...
```
has_value -> true
apply -> 42
next -> 44
a = a + 42
has_value -> false
...

I'll change that.
```

Reply:
Found a problem with this: if the block has an explicit terminator (cont, break, ret), I'd be emitting the "next" after that, which breaks the syntax (basic block with terminator in the middle and not at the end) and semantics (I could end up in a loop that never increments).

I'll keep this behaviour for now and we can worry about the side-effects when we get there.